src/som/interpreter/actors/ResolvePromiseNode.java

```diff
@@ -37,23 +37,23 @@ public SResolver selfResolution(final SResolver resolver, final SPromise result,
   /**
    * Handle the case that a promise is resolved with another promise, which is not itself.
    */
-  @Specialization(guards = {"resolver.getPromise() != result"})
-  public SResolver chainedPromise(final VirtualFrame frame,
-      final SResolver resolver, final SPromise result,
+  @Specialization(guards = {"resolver.getPromise() != resolvedValue"})
+  public SResolver chainedPromise(final VirtualFrame frame, final SResolver resolver, final SPromise resolvedValue,
```

Comment:
`resolvedValue` isn't a good name.
This is a promise, so even `resolutionValue` sounds strange and is unclear later in the code.